### PR TITLE
fix ippool status statistics

### LINKED ISSUE #3391: The data for the used IP is incorrect in the IPPOOL detail page
**Describe the Bug**
**steps**
1. The number of IPs that Ippool has used is 2
2. Delete one of the PODS that uses this ippool
3. In the Ippool list, click on the Ippool to go to the details page
4. View the number of used IPs in the workspace page

**Expected results:**
The number of used IPs should be 1

**Actual results:**
The number of used IPs is 2

https://user-images.githubusercontent.com/36271543/108972903-bc4da100-76be-11eb-8944-3451799ce775.mp4

**ks version:**
nightly-20210222

/kind bug
/assign @duanjiong
/priority Medium
/milestone 3.1
/area networking

### LINKED ISSUE #3393: Select ippool to create service. POD IP is not correct after creation
**Describe the Bug**
**steps**

1. To create a stateless service, in Advanced Settings, click on the Basic Network and select the assigned Ippool-A
2. click create button
3. After the service creation is complete, look at the IP of the service‘s POD

**Expected results:**
The IP of POD belongs to Ippool -A

**Actual results:**
The IP of pod belongs to the default ippool

ks version：
nightly-20210222

/kind bug
/assign @duanjiong 
/priority High
/milestone 3.1.0
/area networking

### PR DESCRIPTION
**Pull request by duanjiong**
Signed-off-by: Duan Jiong <djduanjiong@gmail.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
1. fix ippool status statistics
2. and delete ippool label while workspace is deleted
3. sync default ippool to namespace annotation

**Which issue(s) this PR fixes**:

Fixes #3391 #3386 #3390

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/network/ippool/ippool_controller.go [hunk 1]
```diff
@@ -116,12 +121,15 @@ func (c *IPPoolController) removeFinalizer(pool *networkv1alpha1.IPPool) error {
...
 	b := obj.(*networkv1alpha1.IPPool)
-	_, cidr, err := cnet.ParseCIDR(b.Spec.CIDR)
+	ip, cidr, err := cnet.ParseCIDR(b.Spec.CIDR)
 	if err != nil {
 		return fmt.Errorf("invalid cidr")
 	}
 
 	size, _ := cidr.Mask.Size()
+	if ip.IP.To4() != nil && size == 32 {
+		return fmt.Errorf("the cidr mask should not be 32")
```
Comment (zryfish) [inline pkg/controller/network/ippool/ippool_controller.go:131]:
```suggestion
		return fmt.Errorf("the cidr mask must be less than 32")
```

File: pkg/controller/network/ippool/ippool_controller.go [hunk 2]
```diff
@@ -61,38 +65,39 @@ type IPPoolController struct {
 	ippoolSynced   cache.InformerSynced
 	ippoolQueue    workqueue.RateLimitingInterface
 
+	wkInformer tenantv1alpha1informers.WorkspaceInformer
```
Comment (zryfish) [inline pkg/controller/network/ippool/ippool_controller.go:68]: normally, we use `ws` as an abbreviation for workspace

File: pkg/controller/network/ippool/ippool_controller.go [hunk 3]
```diff
@@ -163,6 +171,27 @@ func (c *IPPoolController) ValidateCreate(obj runtime.Object) error {
...
+			labels.Set{
+				networkv1alpha1.IPPoolDefaultLabel: "",
+			}).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	poolLen := len(pools.Items)
+	if poolLen > 1 || poolLen == 0 {
```
Comment (zryfish) [inline pkg/controller/network/ippool/ippool_controller.go:186]:
```suggestion
	if poolLen != 1 || pools.Items[0].Name != p.Name {
	    return nil
	}
```

File: pkg/controller/network/ippool/ippool_controller.go [hunk 4]
```diff
@@ -193,6 +231,11 @@ func (c *IPPoolController) ValidateDelete(obj runtime.Object) error {
 		return fmt.Errorf("ippool is in use, please remove the workload before deleting")
 	}
 
+	err := c.validateDefaultIPPool(p)
```
Comment (zryfish) [inline pkg/controller/network/ippool/ippool_controller.go:234]:
```suggestion
    return c.validateDefaultIPPool(p)
```

**Comment by zryfish** [comment]
@duanjiong update?

File: pkg/controller/network/ippool/ippool_controller.go (hunk 4 above)
Comment (duanjiong) [reply pkg/controller/network/ippool/ippool_controller.go]: done

File: pkg/controller/network/ippool/ippool_controller.go (hunk 2 above)
Comment (duanjiong) [reply pkg/controller/network/ippool/ippool_controller.go]: done

File: pkg/controller/network/ippool/ippool_controller.go (hunk 1 above)
Comment (duanjiong) [reply pkg/controller/network/ippool/ippool_controller.go]: done

File: pkg/controller/network/ippool/ippool_controller.go (hunk 3 above)
Comment (duanjiong) [reply pkg/controller/network/ippool/ippool_controller.go:186]: The purpose here is to ensure that there will be at least one default ippool, but when there is no default ippool in etcd, creating an ippool will not fail.

Comment (zryfish) [reply pkg/controller/network/ippool/ippool_controller.go:186]: yeah, the suggested change is equal to the code below but simpler

```
if poolLen > 1 || poolLen == 0 {
    return nil
} else if poolLen == 1 && pools.Items[0].Name != p.Name {
    return nil
}
```

Comment (duanjiong) [reply pkg/controller/network/ippool/ippool_controller.go:186]: i see

**Pull request merged**